### PR TITLE
feat(network): add field expose_routes_to_vswitch

### PR DESCRIPTION
**Pull request by jooola**

### CODE REVIEW

File: hcloud/networks/client.py [hunk 1]
```diff
@@ -229,48 +230,66 @@ def create(
...
+            for subnet in subnets:
+                data_subnet = {
                     "type": subnet.type,
                     "ip_range": subnet.ip_range,
                     "network_zone": subnet.network_zone,
                 }
-                for subnet in subnets
-            ]
+                if subnet.vswitch_id is not None:
+                    data_subnet["vswitch_id"] = subnet.vswitch_id
```
Comment (jooola) [inline hcloud/networks/client.py:249]: According to the docs, the vswitch_id field is required when the subnet type is a vswitch. Should we leave this to the user, or do we want to do some validation in the library ?

Comment (apricote) [reply hcloud/networks/client.py:249]: There is always the risk that the validation gets "out-of-sync" with the API, which causes frustration with the user. I would only add client side validation if it substantially improves the user experience.

File: hcloud/networks/client.py [hunk 2]
```diff
@@ -229,48 +230,68 @@ def create(
...
         :param labels: Dict[str, str] (optional)
                User-defined labels (key-value pairs)
         :return: :class:`BoundNetwork <hcloud.networks.client.BoundNetwork>`
         """
         data = {}
         if name is not None:
             data.update({"name": name})
+
+        if expose_routes_to_vswitch is not None:
+            data["expose_routes_to_vswitch"] = expose_routes_to_vswitch
```
Comment (apricote) [inline hcloud/networks/client.py:290]: The surrounding lines are using `data.update(dict)`, is there a reason to prefer one over another?

**Review by apricote — CHANGES_REQUESTED** [review]

File: hcloud/networks/client.py (hunk 2 above)
Comment (jooola) [reply hcloud/networks/client.py:290]: The existing method is somewhat overkill, you have to create a new dict with the value you want to insert, and merge it into the existing dict, probably by iterating over incoming data.

The method I used is simply adding a value at the defined key.

I can use the existing method, this doesn't have a huge impact.

**Review by apricote — APPROVED** [review]

**Pull request merged**